nestkernel/conn_builder.cpp

```diff
@@ -850,3 +1318,166 @@ nest::BernoulliBuilder::connect_()
+ * @param conn_spec connectivity specs
+ * @param syn_spec synapse specs
+ */
+nest::SPBuilder::SPBuilder( Network& net,
+  const GIDCollection& sources,
+  const GIDCollection& targets,
+  const DictionaryDatum& conn_spec,
+  const DictionaryDatum& syn_spec )
+  : ConnBuilder( net, sources, targets, conn_spec, syn_spec )
+{
+  // Check that both pre and post synaptic element are provided
+  if ( pre_synaptic_element_name == "" || post_synaptic_element_name == "" )
+  {
+    throw BadProperty( "pre_synaptic_element and/or post_synaptic_elements is missing" );
+  }
+}
+
+/**
+ * Makes sure that the min delay has been configured correctly for the creation
+ * of new synapses
+ * @return the min_delay configured for the dynamically created synapses
+ */
+nest::Time
+nest::SPBuilder::get_min_delay() const
+{
+  Time min_delay;
+  DictionaryDatum syn_defaults;
+  bool delay = get_default_delay();
+  syn_defaults = net_.get_connector_defaults( get_synapse_model() );
+  if ( !delay ) // delay can be random so min_delay must be defined in the synapse prototype
+  {
+    min_delay = Time::ms( getValue< double_t >( syn_defaults, "min_delay" ) );
+    if ( min_delay == Time::pos_inf() )
+    {
+      throw BadProperty(
+        "Structural Plasticity: to use different delays for the synapses you must specify the min "
+        "and max delay in the synapse default parameters." );
+    }
+  }
+  else // we used the default delay
+  {
+    min_delay = Time::ms( getValue< double_t >( syn_defaults, "delay" ) );
+  }
+  return min_delay;
+}
+
+/**
+ * Makes sure that the max delay has been configured correctly for the creation
+ * of new synapses
+ * @return the max_delay configured for the dynamically created synapses
+ */
+nest::Time
+nest::SPBuilder::get_max_delay() const
+{
+  Time max_delay;
+  DictionaryDatum syn_defaults;
+  bool delay = get_default_delay();
+  syn_defaults = net_.get_connector_defaults( get_synapse_model() );
+  if ( !delay ) // delay can be random so max_delay must be defined in the synapse prototype
+  {
+    max_delay = Time::ms( getValue< double_t >( syn_defaults, "max_delay" ) );
+    if ( max_delay == Time::neg_inf() )
+    {
+      throw BadProperty(
+        "Structural Plasticity: to use different delays for the synapses you must specify the min "
+        "and max delay in the synapse default parameters." );
+    }
+  }
+  else // we used the default delay
+  {
+    max_delay = Time::ms( getValue< double_t >( syn_defaults, "delay" ) );
+  }
+  return max_delay;
+}
+
+void
+nest::SPBuilder::connect( GIDCollection sources, GIDCollection targets )
+{
+  connect_( sources, targets );
+
+  // check if any exceptions have been raised
+  for ( thread thr = 0; thr < net_.get_num_threads(); ++thr )
+    if ( exceptions_raised_.at( thr ).valid() )
+      throw WrappedThreadException( *( exceptions_raised_.at( thr ) ) );
+}
+
+void
+nest::SPBuilder::connect_()
+{
+  connect_( sources_, targets_ );
+}
```

Comment:
I cannot see any place, where this `protected` method will be called. Best would be to to indicate this by throwing a `NotImplemented` exception, or such (maybe even in the header).